htdocs/modules/system/themes/modern/modern.php

```
@@ -0,0 +1,435 @@
+                if (isset($info['icon_big'])) {
+                    $rtn['icon'] = XOOPS_URL . '/modules/' . $mod->getVar('dirname', 'n') . '/' . $info['icon_big'];
+                } elseif (isset($info['image'])) {
+                    $rtn['icon'] = XOOPS_URL . '/modules/' . $mod->getVar('dirname', 'n') . '/' . $info['image'];
+                }
+                $modules_list[] = $rtn;
+            }
+        }
+
+        $tpl->assign('module_menu', $module_menu);
+        $tpl->assign('modules', $modules_list);
+    }
```

PHPMD (2.15.0)

[warning] The method buildMenu() has a Cyclomatic Complexity of 18. The configured cyclomatic complexity threshold is 10. (undefined)

(CyclomaticComplexity)